examples/corpus_token_model.rs

```diff
@@ -43,51 +48,61 @@ pub fn main() {
     }
   };
   let mut token_writer = BufWriter::with_capacity(10485760, token_model_file);
-  let space = b" ";
-  let linebreak = b"\n";
+  let space = ' ';
+  let linebreak = '\n';
+  // Integers, floats, subfigure numbers
+  let is_numeric = Regex::new(r"^-?(?:\d+)(?:[a-k]|(?:\.\d+(?:[eE][+-]?\d+)?))?$").unwrap();
 
   let mut corpus = Corpus::new(corpus_path);
   for mut document in corpus.iter() {
     document_count += 1;
     for mut paragraph in document.paragraph_iter() {
       paragraph_count += 1;
       for mut sentence in paragraph.iter() {
-        sentence_count += 1;
-        let mut has_words = false;
-        for word in sentence.simple_iter() {
+        let mut sentence_buffer = String::new();
+        let mut invalid_sentence = true;
+         'words: for word in sentence.simple_iter() {
           if !word.range.is_empty() {
             let mut word_string = word.range.get_plaintext().to_lowercase();
-            utf_truncate(&mut word_string, 50);
-            if word_string == "mathformula" {
+            if word_string.len() > 30 { 
+              // Using a more aggressive normalization, large words tend to be conversion errors with lost whitespace - drop the entire sentence when this occurs.
+              overflow_count += 1;
+              invalid_sentence = true;
+              break 'words;
+            }
+            let mut word_str : &str = &word_string;
+            // Note: the formula and citation counts are an approximate lower bound, as sometimes they are not cleanly tokenized, e.g.
+            // $k$-dimensional will be the word string "mathformula-dimensional"
+            if word_string.contains("mathformula") {
+              word_str = "mathformula";
```

Comment:
I should note that the normalization is aggressive here since most of the inaccuracies come from tokenization failures, which in turn come from weird / inconsistent uses of latex math mode in arXiv. 

For example: `e$^2$o` or some such. A perfectly purified separation of the modalities as in #3 would be a good upgrade to have here, for now erring on the side of quality.